### PR TITLE
🐛 Guard nextNewNFTId if NFT has expired

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

**Comment by WeiJun0827** [comment]
Hmmm... actually the expiry time has been guarded in `getPriceInfo` if someone tries to purchase an expired NFT. And the `nextNewNFTId` was intact before.
https://github.com/likecoin/likecoin-api-public/blob/master/src/util/api/likernft/purchase.ts#L166-L168
Should we set `nextNewNFTId` so that [line 160](https://github.com/likecoin/likecoin-api-public/blob/master/src/util/api/likernft/purchase.ts#L160) will throw a `SELLING_NFT_DOC_NOT_FOUND` error?

**Comment by WeiJun0827** [comment]
Another off-the-topic concern is that I found `getLatestNFTPriceAndInfo` is called twice for each NFT in `POST /purchase` API.
It is [called](https://github.com/likecoin/likecoin-api-public/blob/master/src/routes/likernft/purchase.ts#L72) to calculate and check the granted amount the first time.
And [called again](https://github.com/likecoin/likecoin-api-public/blob/master/src/util/api/likernft/purchase.ts#L159) to process the purchase in `processNFTPurchase` -> `getPriceInfo`.

Should we move the check of the first step into the second one to prevent race condition/inconsistency?

File: src/util/api/likernft/purchase.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ export async function getLatestNFTPriceAndInfo(
     price = currentPrice;
     // This NFT ID represents a possible NFT of that NFT Class for purchasing only,
     // another fresh one might be used on purchase instead
-    nextNewNFTId = newNftDocData.id;
+    if (!collectExpiryAt || collectExpiryAt < Date.now()) nextNewNFTId = newNftDocData.id;
     isProcessing = newNftDocData.isProcessing;
```
Comment (williamchong) [inline src/util/api/likernft/purchase.ts:141]: maybe we should not set anything if `collectExpiryAt` is expired, act like there is no `newNftDocData`

**Review by williamchong — CHANGES_REQUESTED** [review]

**Comment by williamchong** [comment]
> Another off-the-topic concern is that I found `getLatestNFTPriceAndInfo` is called twice for each NFT in `POST /purchase` API. It is [called](https://github.com/likecoin/likecoin-api-public/blob/master/src/routes/likernft/purchase.ts#L72) to calculate and check the granted amount the first time. And [called again](https://github.com/likecoin/likecoin-api-public/blob/master/src/util/api/likernft/purchase.ts#L159) to process the purchase in `processNFTPurchase` -> `getPriceInfo`.
> 
> Should we move the check of the first step into the second one to prevent race condition/inconsistency?

The original idea was to check the tx info and the price matches before going into more expensive db transaction, we should no rely on  any information provided by first check in the later sections.

File: src/util/api/likernft/purchase.ts (hunk 1 above)
Comment (WeiJun0827) [reply src/util/api/likernft/purchase.ts:141]: OK, I should close this PR

**Pull request closed without merging**